src/main/java/uk/gov/digital/ho/pttg/application/namematching/candidates/AliasCombinations.java

```diff
@@ -8,15 +9,17 @@
 import java.util.List;
 
 import static uk.gov.digital.ho.pttg.application.namematching.candidates.AliasCombinationsFunctions.*;
+import static uk.gov.digital.ho.pttg.application.namematching.candidates.NameMatchingCandidateGenerator.ALIAS_COMBINATIONS_STRATEGY_PRIORITY;
 
 @Component
+@Order(value = ALIAS_COMBINATIONS_STRATEGY_PRIORITY)
 public class AliasCombinations implements NameMatchingCandidateGenerator {
 
     @Override
     public List<CandidateName> generateCandidates(InputNames inputNames) {
         List<CandidateName> candidateNames = new ArrayList<>();
 
-        if (inputNames.aliasSurnames().isEmpty()) {
+        if (!inputNames.hasAliasSurnames()) {
```

Comment:
I like the idea of calling all the generators, rather than having the service select them. But we need a unit test that AliasCombinations returns an empty list when there are no Aliases and, similarly NameCombinations returns an empty list when there are aliases.

Reply:
In the case of AliasCombinations, there was an existing test for this: shouldReturnEmptyListWhenNoAliasSurnames

Reply:
In the case of NameCombinations, I had already added one: shouldReturnEmptyListWhenAliasSurnames